c/datasetjson.c

```diff
@@ -2251,9 +2254,238 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
 #endif /* __ZOWE_OS_ZOS */
 }
 
+/*************************/
+#define FIELD_MAX_LENGTH    1024
+#define REQUEST_MAX_LENGTH  2048
 
-#endif /* not METTLE - the whole module */
+int qsamAllocFile( qsamFileRequest *request, char* message, 
+                   int messageLength){
+#ifdef __ZOWE_OS_ZOS
+  char *baseRequest = "alloc new catalog msg(2) ";
+  char *freeRequest = "free  ";
+  char tempString[20] = {0};
+
+  char field[FIELD_MAX_LENGTH];
+  int status, value;
+  char *ptr;
+
+  /* Must supply daName */
+  if (request->daName == NULL) {
+    return -1;
+  }
+
+  /* Allocate structure in A31 space */
+  EXTF *bpxwdyn=(EXTF *)fetch("BPXWDYN");
+  ALLOC_STRUCT31(
+    STRUCT31_NAME(below2G),
+    STRUCT31_FIELDS(
+      char  request[REQUEST_MAX_LENGTH];
+    )
+  );
+
+  /* setup initial request string */
+  strncpy( below2G->request, baseRequest, sizeof(below2G->request));
+
+  /* Initialize the DD/FI field */
+  ptr = (request->ddName == NULL ?  QSAM_DEFAULT_DDNAME: request->ddName);
```

Comment:
We already have code for dynalloc in https://github.com/zowe/zowe-common-c/blob/master/c/dynalloc.c, you should re-use/enhance code from there.